packages/components/tests/dummy/app/templates/components/form/text-input.hbs

```diff
@@ -0,0 +1,478 @@
+          <code class="dummy-code">controlId</code>
+          value of the control.</em></p>
+    </dd>
+    <dt>&lt;[C].HelperText&gt; <code>yielded component</code></dt>
+    <dd>
+      <p>It is a container that yields its content inside the "helper text" block.</p>
+      <p>The content can be a simple string, or a more complex/structured one (in which case it inherits the text
+        style).</p>
+      <p>For details about its API check the
+        <LinkTo @route="components.form.base-elements"><code class="dummy-code">Form::HelperText</code></LinkTo>
+        component.</p>
+      <p><em>Notice: the
+          <code class="dummy-code">id</code>
+          attribute of the element is automatically generated, using the
+          <code class="dummy-code">controlId</code>
+          value of the control.</em></p>
+    </dd>
+    <dt>&lt;[C].Error&gt; <code>yielded component</code></dt>
+    <dd>
+      <p>It is a container that yields its content inside the "error" block.</p>
+      <p>The content can be a simple string, or a more complex/structured one (in which case it inherits the text
+        style).</p>
+      <dl class="dummy-component-props">
+        <dt>[E].Message <code>yielded component</code></dt>
+        <dd>
+          <p>If the error is made of multiple messages, you can iterate over a collection of error messages yielding
+            individual items using
+            <code class="dummy-code">Error.Message</code>.
+          </p>
+        </dd>
+      </dl>
+      <p>For details about its API check the
+        <LinkTo @route="components.form.base-elements"><code class="dummy-code">Form::Error</code></LinkTo>
+        component.</p>
+      <p><em>Notice: the
+          <code class="dummy-code">id</code>
+          attribute of the
+          <code class="dummy-code">Error</code>
+          element is automatically generated.</em></p>
+    </dd>
+  </dl>
+
+</section>
+
+<section>
+  <h3 class="dummy-h3" id="how-to-use"><a href="#how-to-use" class="dummy-link-section">§</a> How to use</h3>
+  <p class="dummy-paragraph">🚧 TODO 🚧</p>
+</section>
+
+<section>
+  <h3 class="dummy-h3" id="design-guidelines"><a href="#design-guidelines" class="dummy-link-section">§</a>
+    Design guidelines</h3>
+  <p class="dummy-paragraph">🚧 TODO 🚧</p>
+  {{! UNCOMMENT THIS BLOCK (once the link and/or the image are available) }}
+  {{!
+  <div class="dummy-design-guidelines">
+    <p class="dummy-paragraph">
+      <a href="[ADD THE LINK TO THE FIGMA FILE/PAGE HERE!]" target="_blank" rel="noopener noreferrer">Figma UI Kit</a>
+    </p>
+    <br />
+    <img class="dummy-figma-docs" src="/assets/images/form-text-input-design-usage.png" alt="" role="none" />
+  </div>
+  }}
+</section>
+
+<section>
+  <h3 class="dummy-h3" id="accessibility"><a href="#accessibility" class="dummy-link-section">§</a> Accessibility</h3>
+  <p class="dummy-paragraph">🚧 TODO 🚧</p>
+</section>
+
+<section data-test-percy>
```

Comment:
Continuing the conversation from this PR: https://github.com/hashicorp/design-system/pull/285/files#r896982292

---

As discussed during the code review [the other day], I am not convinced this is the way to go for the "showcase" components. I totally understand the idea to have accessible documentation pages, but the _show_case is intended mainly to _show_ the components, with a particular focus on the developers implementing the components (and Percy) plus as a bonus side-effect, show to the consumers what are the variants (and their combinations) of a component. 

The issue of devs copying&pasting the code from the showcase is minimal, because we don't expect them to copy the HTML, and (unless I'm missing something) is not possible to copy the code from the Ember inspector.

So the last problems remains having pages which will not pass an accessibility test. Here I can propose some alternatives:
- we hide the whole showcase from the accessibility tree (a bit drastic!)
- we hide the parent container/wrapper of the "base" components from the accessibility tree (more specific)
- we accept that not all the elements in the showcase will meet all the accessibility criteria, when considered in isolation
  - this is not something unusual: consider this MDN page: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input - as you can see the top example has a label associated with the `<input>` but the ones in the list of types they don't and they also don't have a `aria-label` associated, and I am not sure I would expect them to have it frankly (their purpose meant to showcase the component, not as real usable/interactive content).

Reply:
I did a quick test on other design systems websites that we use as reference, these are the results for similar pages:
<img width="645" alt="screenshot_1572" src="https://user-images.githubusercontent.com/686239/174783746-b7326632-ac50-450e-bda0-3449ff58e5f0.png">
<img width="641" alt="screenshot_1571" src="https://user-images.githubusercontent.com/686239/174783752-c8b955e5-62b8-4e5f-b62b-4059b0c6d3ff.png">
<img width="643" alt="screenshot_1570" src="https://user-images.githubusercontent.com/686239/174783756-d62fce00-2131-40b5-89e1-fdeb82963932.png">
<img width="643" alt="screenshot_1569" src="https://user-images.githubusercontent.com/686239/174783760-54fc5019-cc3a-49d4-98de-d5000dfb6862.png">

Reply:
Let's forget that this is something that matters to me (an equal member of the team), and look over the other reasons why this should be done. 

It's still inconsistent from the rest of our documentation pages and our approach so far.

I'm inferring two things from your response: 
1. someone with a screen reader will never need to use our documentation
2. other design systems have non-conformant documentation (so it's fine)

I think the first is perhaps the reason that we have to have accessibility laws in place-  what I have asked for is:
1. a small thing to fix 
2. a thing that will make our page as consistent as our other pages are
3. a thing that demonstrates our accessibility statement: intent to provide at least WCAG conformance

Even if I were to accept the argument that "other design systems have conformance issues in their documentation too", then compare the number of critical issues, because that's what this is. 

What would it mean then that a person who needs a screen reader will never visit this page?
1. that we would never hire someone who needed a screen reader
2. that none of us will never need a screen reader
3. that no auditor will ever be using a screen reader in their audit
4. that no site visitor (maybe works on another design system and has come to ours to see how we do things) will ever be using a screen reader

I don't think any of those things are logically safe to assume. 

However, even if I were to accept that a person who needs a screen reader were to never access this documentation, I have to wonder why you're so resistant to this. All of our other pages are conformant. This clearly is a critical WCAG issue. 

I'll even submit a PR to fix it.

Reply:
I opened a ticket in our backlog to track this discussion